tests: drivers: dma: configure alignment based on devicetree

alexapostolu opened the pull request:
Configure data buffer alignment based on devicetree `dma_buf_addr_alignment` property.

Fixes #98949

cfriedt reviewed (dismissed):
Just a nit (non-blocking) - otherwise LGTM.

JarmouniA commented on tests/drivers/dma/chan_blen_transfer/src/test_buffers.c:
```diff
@@ -8,5 +8,6 @@
-__aligned(32) char rx_data[TEST_BUF_SIZE] = { 0 };
+__aligned(CONFIG_DMA_BLEN_ALIGNMENT) char tx_data[TEST_BUF_SIZE] =
+	"It is harder to be kind than to be wise........";
+__aligned(CONFIG_DMA_BLEN_ALIGNMENT) char rx_data[TEST_BUF_SIZE] = {0};
```
```suggestion
static __aligned(CONFIG_DMA_BLEN_ALIGNMENT) char rx_data[TEST_BUF_SIZE] = {0};
```

teburd reviewed (changes requested):
I think its fine to actually make this configurable... but changing this here needs more time to cook than the next release. It's broken many end users in the past mucking around with things here. @hakehuang and NXP in particular. I want more time for this to cook than the next release. This isn't fixing any application facing bugs, its fixing a test case issue.

JarmouniA commented on tests/drivers/dma/chan_blen_transfer/src/test_buffers.c:
```diff
@@ -8,5 +8,6 @@
 
-__aligned(32) char tx_data[TEST_BUF_SIZE] = "It is harder to be kind than to be wise........";
-__aligned(32) char rx_data[TEST_BUF_SIZE] = { 0 };
+__aligned(CONFIG_DMA_BLEN_ALIGNMENT) char tx_data[TEST_BUF_SIZE] =
```
Who should have the priority, DCache line size of DMA-required alignment? (or just take the bigger one I guess)

alexapostolu commented:
Thanks for the feedback, I can make the alignment use the DT attribute instead of a KConfig, and have it ready for 4.4.0

alexapostolu committed:
tests: drivers: dma: use tst_dma0 across all tests

Use tst_dma0 node across all tests for consistency.

Signed-off-by: Alex Apostolu <aapostolu@tenstorrent.com>

alexapostolu committed:
tests: drivers: dma: configure alignment based on devicetree

Configure the alignment of the data buffers based on the
dma_buf_addr_alignment property. Default to 32 if property isn't set.

Signed-off-by: Alex Apostolu <aapostolu@tenstorrent.com>

alexapostolu commented:
@teburd @JarmouniA This is ready for another review, thanks

hakehuang reviewed (commented):
for chan_link_transfer case, need add mimxrt595_evk_mimxrt595s_cm33.overlay with the tst_dma dts node support.

hakehuang reviewed (commented):
mimxrt595 does not use `edma0` but `dma0`

JarmouniA commented on tests/drivers/dma/scatter_gather/testcase.yaml:
```diff
@@ -21,7 +21,7 @@ tests:
       - adp_xc7k/ae350/clic
       - bg29_rb4420a
-    filter: dt_alias_exists("dma0")
+    filter: dt_nodelabel_enabled("tst_dma0")
```
ditto

cfriedt replied:
That's generally OK and pretty common for testsuites. I don't see it as being problematic.

JarmouniA commented on tests/drivers/dma/cyclic/src/test_dma_cyclic.c:
```diff
@@ -48,7 +48,7 @@ static int test_cyclic(void)
 	}
 
-	dma = DEVICE_DT_GET(DT_ALIAS(dma0));
+	dma = DEVICE_DT_GET(DT_NODELABEL(tst_dma0));
```
ditto

JarmouniA commented on tests/drivers/dma/scatter_gather/src/test_dma_sg.c:
```diff
@@ -66,7 +66,7 @@ static int test_sg(void)
 	memset(rx_data, 0, sizeof(rx_data));
 
-	dma = DEVICE_DT_GET(DT_ALIAS(dma0));
+	dma = DEVICE_DT_GET(DT_NODELABEL(tst_dma0));
```
> Although is there a reason the scatter_gather and cyclic uses an alias but the other test suites use the node label?

No specific reason really.

alexapostolu replied:
Okay I can do that. But isn't it better to have the `tst_dma0: &dma0 {}` label in a overlay inside the `tests/` directory rather than in the DTS?

hakehuang reviewed (dismissed):
all NXP test is clean

alexapostolu commented:
Rebased to fix merge conflits

teburd reviewed (approved):
Good changes, thanks!
